portfolio/src/main/java/com/google/sps/servlets/EntityExtractionServlet.java

```diff
@@ -0,0 +1,73 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.cloud.language.v1.Document;
+import com.google.cloud.language.v1.LanguageServiceClient;
+import com.google.cloud.language.v1.Sentiment;
+import com.google.cloud.language.v1.AnalyzeEntitiesRequest;
+import com.google.cloud.language.v1.AnalyzeEntitiesResponse;
+import com.google.cloud.language.v1.Document.Type;
+import com.google.cloud.language.v1.Entity;
+import com.google.cloud.language.v1.EncodingType;
+import java.io.IOException;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import com.google.gson.Gson;
+import java.util.ArrayList;
+
+@WebServlet("/entity")
+public class EntityExtractionServlet extends HttpServlet {
+
+  public final static String ENTITY_AREA = "entity-comment";
+  String message;
+  ArrayList<String> entities = new ArrayList<String>();
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
```

Comment:
Rather than splitting this into a POST method for getting the message input and running the entity extraction and a GET method for returning the result, there should be a single method so that the JS can make one request and receive the response to place into the page.

Because that method will not modify state on the server, the combined method should be a GET.

Same idea for sentiment analysis.

With the two method approach, storing the data in a variable whose value persists across requests means that when I load the page initially, I will see the entities extracted from the previous user's input. And if two users send a request at almost the same time, one may get the entities from the other's input, since their entities have been overwritten, or if this code is executing simultaneously for both user requests, a [ConcurrentModificationException](https://docs.oracle.com/javase/7/docs/api/java/util/ConcurrentModificationException.html) could be thrown when both threads are trying to modify the `entities` list simultaneously.

Also, making this a single method should simplify the Java and JS code, as well as improving the speed of user experience because the browser won't have to do a full page reload to show the result.